app/models/sample_attribute.rb

```diff
@@ -32,6 +32,10 @@ def required?
     super || is_title?
   end
 
+  def pre_process_value(value)
+    sample_attribute_type.pre_process_value(value, controlled_vocab: sample_controlled_vocab, linked_sample_type: linked_sample_type)
+  end
+
```

Comment:
is this method needed? I was recently removed thinking it was now unused - I wonder if it reappeared due to a merge problem